extension/src/experiments/workspace.ts

```diff
@@ -28,13 +29,15 @@ export class WorkspaceExperiments extends BaseWorkspaceWebviews<
   public readonly updatesPaused: EventEmitter<boolean>
 
   private readonly checkpointsChanged: EventEmitter<void>
+  private readonly hasDataChanged: EventEmitter<void>
```

Comment:
Might make things easier if you listen to `WorkspaceExperiments.columnsChanged.event` in the Extension instead of injecting this in. (or create `onDidChangeExperimentColumns` and listen to that)